src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -1433,6 +1437,9 @@ private static string FormatIgnoreCondition(JsonIgnoreCondition ignoreCondition)
             private static string FormatJsonSerializerDefaults(JsonSerializerDefaults defaults)
                 => SourceGeneratorHelpers.FormatEnumLiteral(JsonSerializerDefaultsTypeRef, defaults);
 
+            private static string FormatReferenceHandler(JsonKnownReferenceHandler referenceHandler)
+                => SourceGeneratorHelpers.FormatEnumLiteral(ReferenceHandlerTypeRef, referenceHandler);
```

Comment:
Not saying that this wouldn't work, but `ReferenceHandler` isn't technically an enum. It's a class with static properties, so perhaps different concerns apply to formatting it?

Reply:
The default value of the enum didn't map to a default reference handler in the first iteration of this PR. Updated the PR to handle this enum -> reference handler object conversion. Note that for the naming policy, we actually have the policy classes in the Common folder so `nameof` works [there](https://github.com/dotnet/runtime/blob/a3fe47ef1a8def24e8d64c305172199ae5a4ed07/src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs#L1269-L1277). I tried to see if I could also move `ReferenceHandler` to Common but it has a couple of dependencies so I just left it as it is instead of adding a bunch of types to Common. I just hardcoded the property names for this PR.